File: app/models/commitment.py
import datetime


class Commitment:
    def __init__(self, date: datetime.date, user_id: str, user_name: str, time: str):
        self.date = date
        self.user_id = user_id
        self.user_name = user_name
        self.time = time

    def __str__(self):
        return f"Commitment(date={self.date}, user_id={self.user_id}, user_name={self.user_name}, time={self.time})"


class UserCommitment:
    def __init__(
        self, user_id: str, user_name: str, time: str, dates: list[datetime.date]
    ):
        self.user_id = user_id
        self.user_name = user_name
        self.time = time
        self.dates = dates

    def __repr__(self):
        return f"UserCommitment(user_id={self.user_id}, user_name={self.user_name}, time={self.time}, dates={self.dates})"

File: frontend/lib/models/commitment.dart
class Commitment {
  final DateTime? date;
  final String? userId;
  final String? userName;
  final String? time;

  Commitment({
    this.date,
    this.userId,
    this.userName,
    this.time,
  });

  @override
  String toString() {
    return "Commitment: $date, $userId, $userName, $time";
  }
}

class UserCommitment {
  final String? userId;
  final String? userName;
  final String? time;
  final List<DateTime>? dates;

  UserCommitment({
    this.userId,
    this.userName,
    this.time,
    this.dates,
  });

  @override
  String toString() {
    return "UserCommitment: $userId, $userName, $time, $dates";
  }
}
